File: gateway/src/models/common/entry.go
package common

import "github.com/google/uuid"

type EntityType string // 认证实体类型，表示认证对象的类别，可以是用户、设备或服务等。
type UserRole string
type DeviceStatus string

const (
	UserRoleAdmin UserRole = "admin"
	UserRoleUser  UserRole = "user"
)

const (
	EntityUser    EntityType = "user"
	EntityDevice  EntityType = "device"
	EntityService EntityType = "service"
)

const (
	DeviceStatusActive   DeviceStatus = "active"
	DeviceStatusDisabled DeviceStatus = "disabled"
	DeviceStatusBlocked  DeviceStatus = "blocked"
	DeviceStatusInvoked  DeviceStatus = "invoked"
)

// User 代表认证服务器中的用户实体。
type User struct {
	ID           uuid.UUID
	AccountID    string
	UserName     string
	PasswordHash string
	Role         UserRole
}

// EdgeDevice 代表认证服务器中的边缘设备实体。
type EdgeDevice struct {
	ID               uuid.UUID
	Name             string
	DeviceSecretHash string
	Zone             string
	Status           DeviceStatus
	LastOnline       int64
}

// ServiceEntry 代表认证服务器中的服务实体。
type ServiceEntry struct {
	ID                uuid.UUID
	Name              string
	ServiceSecretHash string
	ActiveCommKeyID   string
	KeyEntityType     string
}

File: data_server/src/models/common/entry.py
from typing import Literal
from uuid import UUID

from msgspec import Struct

EntityType = Literal["user", "service", "device"]
UserRole = Literal["admin", "user"]
DeviceStatus = Literal["active", "disabled", "blocked", "invoked"]


class User(Struct, frozen=True):
    """认证用户实体。"""

    id: UUID
    account_id: str
    user_name: str
    password_hash: str
    role: UserRole


class EdgeDevice(Struct, frozen=True):
    """边缘设备实体。"""

    id: UUID
    name: str
    device_secret_hash: str
    zone: str
    status: DeviceStatus
    last_online: int


class ServiceEntry(Struct, frozen=True):
    """服务实体定义。"""

    id: UUID
    name: str
    service_secret_hash: str
    active_comm_key_id: str
    key_entity_type: str
